src/pointer/ptr.rs

```diff
@@ -0,0 +1,1010 @@
+use core::ptr::NonNull;
+
+use crate::{util::AsAddress, KnownLayout, _CastType};
+
+/// Module used to gate access to [`Ptr`]'s fields.
+mod def {
```

Comment:
Are these submodules useful now that all of this code has been moved to `ptr` (as opposed to living directly in `util`?

Reply:
Ack.